### PR TITLE
chgres_cube - Complete removal of wgrib2 library

### LINKED ISSUE #591: Can we remove wgrib2 from the dependencies?
Somehow wgrib2 crept into our dependencies.

We are trying to eliminate use of wgrib2 in NCEPLIBS. The wgrib2 library is not well-maintained, distributed, or tested, and there is no interest on the wgrib2 team in meeting our needs in these areas. So we need to not build more tools on top of wgrib2.

### PR DESCRIPTION
**Pull request by GeorgeGayno-NOAA**
## DESCRIPTION OF CHANGES: 
Replace remaining calls to wgrib2 library with the G2LIB. Specifically, replace
the calls that read the input GRIB2 file and return the grid point latitude and
longitudes in model_grid.F90.

## TESTS CONDUCTED: 
Tests are documented in the issue - #591. See:
- https://github.com/ufs-community/UFS_UTILS/issues/591#issuecomment-1083177654
- https://github.com/ufs-community/UFS_UTILS/issues/591#issuecomment-1083212098
- https://github.com/ufs-community/UFS_UTILS/issues/591#issuecomment-1083234956
- https://github.com/ufs-community/UFS_UTILS/issues/591#issuecomment-1083477629
- https://github.com/ufs-community/UFS_UTILS/issues/591#issuecomment-1083486767
- https://github.com/ufs-community/UFS_UTILS/issues/591#issuecomment-1085084220
- https://github.com/ufs-community/UFS_UTILS/issues/591#issuecomment-1085098722

## DEPENDENCIES:
None.

## DOCUMENTATION:
N/A

## ISSUE: 
Fixes #591

### CODE REVIEW

File: sorc/chgres_cube.fd/model_grid.F90 [hunk 1]
```diff
@@ -609,71 +605,261 @@ subroutine define_input_grid_mosaic(localpet, npets)
...
+
+ print*,"- COMPUTE GRID CELL CORNER COORDINATES."
+ call gdswzd(kgds,1,(ip1_input*jp1_input),-9999.,xpts_corner,ypts_corner,rlon_corner,rlat_corner,nret)
+
+ if (nret /= (ip1_input*jp1_input)) then
+   call error_handler("GDSWZD RETURNED WRONG NUMBER OF POINTS.", 2)
+ endif
+
+ deallocate(xpts_corner, ypts_corner)
+
```
Comment (GeorgeGayno-NOAA) [inline sorc/chgres_cube.fd/model_grid.F90:738]: Note to self, remove diagnostic print.

File: sorc/chgres_cube.fd/model_grid.F90 [hunk 2]
```diff
@@ -609,71 +605,261 @@ subroutine define_input_grid_mosaic(localpet, npets)
...
+ endif
+
+ do j = 1, j_input
+ do i = 1, i_input
+    rlat_diff(i,j) = rlat_corner(i,j) - rlat(i,j)
+    rlon_diff(i,j) = rlon_corner(i,j) - rlon(i,j)
+ enddo
+ enddo
+
+   ncid = 34
```
Comment (GeorgeGayno-NOAA) [inline sorc/chgres_cube.fd/model_grid.F90]: And don't forget to remove the write of this diagnostic file.

**Review by JeffBeck-NOAA — APPROVED** [review]

File: sorc/chgres_cube.fd/model_grid.F90 (hunk 1 above)
Comment (JeffBeck-NOAA) [reply sorc/chgres_cube.fd/model_grid.F90:738]: I think these can be removed now?

File: sorc/chgres_cube.fd/model_grid.F90 (hunk 2 above)
Comment (JeffBeck-NOAA) [reply sorc/chgres_cube.fd/model_grid.F90]: I think this file write can be removed now?

File: sorc/chgres_cube.fd/model_grid.F90 (hunk 1 above)
Comment (GeorgeGayno-NOAA) [reply sorc/chgres_cube.fd/model_grid.F90:738]: Yes. Thanks for the reminder.

**Pull request merged**